controller/gateway/controller_watch.go

```diff
@@ -118,6 +121,27 @@ func (r *Reconciler) listGatewaysForGatewayClass(ctx context.Context, obj client
 	return
 }
 
+// listGatewaysForKonnectExtension is a watch predicate which finds all Gateways
+// that use a GatewayConfiguration that references a specific KonnectExtension.
+func (r *Reconciler) listGatewaysForKonnectExtension(ctx context.Context, ext *konnectv1alpha1.KonnectExtension) []reconcile.Request {
+	gatewayConfigurationsRequests := index.ListObjectsReferencingKonnectExtension(r.Client, &operatorv1beta1.GatewayConfigurationList{})(ctx, ext)
+	gatewayConfigurations := lo.Map(gatewayConfigurationsRequests, func(req reconcile.Request, _ int) operatorv1beta1.GatewayConfiguration {
+		return operatorv1beta1.GatewayConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: req.Namespace,
+				Name:      req.Name,
+			},
+		}
+	})
+	affectedGateways := make([]reconcile.Request, 0)
```

Comment:
```suggestion
	affectedGateways := make([]reconcile.Request, 0, len(gatewayConfigurations))
```
we can preallocate here

Reply:
we cannot as the same `GatewayConfiguration` can be used by multiple `Gateway`s. There is no 1:1 relationship between `Gateway`s and `GatewayConfiguration`s

Reply:
Ok, I see it may not be worth, but it'd be correct too. Let's leave as is it